pkg/cloud/services/networking/port.go

```diff
@@ -187,6 +191,32 @@ func (s *Service) GetOrCreatePort(eventObject runtime.Object, clusterName string
 	return port, nil
 }
 
+func (s *Service) getSubnetIDForFixedIP(subnet *infrav1.SubnetFilter, networkID string) (string, error) {
+	if subnet == nil {
```

Comment:
Is this normal case? if this is not normal, need to return error.

Reply:
It's valid to specify a FixedIP with IPAddress only (no subnet). From the docs https://docs.openstack.org/api-ref/network/v2/index.html?expanded=create-port-detail#ports

```
If you specify only an IP address, OpenStack Networking tries to allocate the IP address if the address is a valid IP for any of the subnets on the specified network.
```

Reply:
Thanks, I studied.